### PR TITLE
github-cli-auth: authenticate gh api graphql under multi-owner App auth

### PR DESCRIPTION
**Pull request by devxoul**
## Background

Follow-up to #531. That PR governs `gh` at the bash boundary: `github-cli-auth`'s `tool.before` hook extracts the target repo from `-R/--repo` or a literal `gh api /repos/{owner}/{repo}` path, mints that repo's GitHub App installation token, and injects it via the env overlay. It deliberately left `gh api graphql` as pass-through (its own comment: *"A non-repo endpoint (`graphql`, `/user`) passes through — -R does not make it repo-scoped, so no mint."*).

But **graphql carries its target repo inside the query body / opaque node IDs — never an inspectable path or flag the parser reads**. So under multi-owner App auth (where there is *no* single seeded `GH_TOKEN` — `ghTokenSeedDecision` → `skip: 'multiple-owners'`), a `gh api graphql` call runs **unauthenticated** and fails with *"gh not authenticated in this shell."*

### Live symptom

On an agent configured with `channels.github.repos = ["typeclaw/typeclaw", "agent-messenger/agent-messenger"]` (two owners → multi-owner App, no seeded token), the agent could **post REST PR reviews fine** (path-based mint worked) but **could not resolve review threads** — `resolveReviewThread` is a GraphQL-only mutation, so it hit the pass-through gap and failed auth. This was the bot's own report on #568:

> I can't resolve the existing review threads from here because the local `gh` CLI session isn't authenticated in this shell.

## Changes

Both scoped to the `github-cli-auth` bundled plugin.

### 1. Honor `-R/--repo` on `graphql` (parser) — `gh-command.ts`

`classifyGhApiSegment` now mints for `gh api graphql ... -R owner/repo`, treating `-R` purely as a TypeClaw auth-context hint.

**Why it's safe** (and not a #531-style mint-for-X-hit-Y): graphql has **no literal endpoint path** to conflict with, unlike `/repos/{owner}/{repo}` (which trips `API_REPO_CONFLICT_REASON`). There's nothing for the flag to disagree with, and the minted token's **installation scope — not the flag — bounds what the call can reach**. `gh` itself ignores `-R` for graphql routing, so the flag is inert to `gh` and serves only as our hint. The repo is still validated against `channels.github.repos[]` by the existing bridge resolver before any mint.

### 2. Reactive `tool.after` nudge — `graphql-auth-nudge.ts`

When a `gh api graphql` call **fails with a GitHub auth-failure signature** AND **no token is seeded** (the multi-owner App case), append an advisory steering the agent to re-run with `-R owner/repo`. Mirrors the bundled `guard` plugin's in-place `result.content` advice pattern (`uncommitted-changes.ts`).

Gated on an **absent token** (`classifyGhToken(GH_TOKEN) === 'none'`) so single-owner App / classic-PAT / fine-grained-PAT setups — where graphql already authenticates — **never** see the nudge. Detection is result-text-only because `ToolAfterEvent` carries no `args`/`origin`; the signature list is kept narrow (`HTTP 401`, `Bad credentials`, `Resource not accessible by integration`, …) so a graphql payload that merely contains "401" in its data doesn't trip it.

### How they compose

1. Agent's first `gh api graphql ...` (no `-R`) → pass-through → unauthenticated → **auth error**.
2. `tool.after` nudge fires → advises `-R owner/repo`.
3. Agent retries `gh api graphql -R owner/repo ...` → new parser branch → **mints + injects** → succeeds.

The nudge would be inert advice without change 1; change 1 would require the agent to already know the convention without the nudge. Together they self-correct.

## What this does NOT do

- No change to REST behavior, classic/fine-grained PAT handling, or the `tool.before` mint path from #531.
- Does not parse GraphQL bodies or decode node IDs to infer the repo — explicitly rejected as brittle and security-sensitive (a mis-parse would mint the wrong owner's token). `-R` is the explicit, auditable hint instead.
- Does not change installation tokens to be repository-scoped at mint time. Today's tokens are installation-wide; narrowing them (`POST .../access_tokens` with `repositories: [...]`) is the strongest defense-in-depth against same-installation breadth and is worth a follow-up, but is out of scope here.

## Tests

- `gh-command.test.ts` — `graphql -R owner/repo` → inject (incl. `--repo=`); `graphql` no-`-R` → pass-through; non-graphql non-repo endpoints (`/user`, `/rate_limit`) with `-R` → still pass-through.
- `graphql-auth-nudge.test.ts` — fires on auth-failure output with no seeded token; matches the App `Resource not accessible by integration` signature; **does not** fire when a token is seeded, on a successful graphql call, on a non-graphql failure, or on non-bash tools; idempotent (no double-append).

`bun run typecheck`, `bun run lint` (0 errors), `bun run format:check` all clean. Full suite green except one pre-existing `resolveSelfPackageJsonPath` failure (asserts the manifest path ends with `typeclaw/package.json`) — a worktree-directory-name artifact, unrelated to this change.

### CODE REVIEW

**Comment by devxoul** [comment]
Fixed in 9d07136. You're right — `Could not resolve to ...` is a node-id/missing-repo resolution error, not an auth rejection, so it could misroute the nudge. Dropped it from `AUTH_FAILURE_SIGNATURES`, leaving only concrete auth strings (`HTTP 401`, `Bad credentials`, `Resource not accessible by integration`, etc.), and added a regression test asserting node-id / missing-repo resolution errors do **not** trigger the nudge.

**Comment by devxoul** [comment]
@typeey review again

File: src/bundled-plugins/github-cli-auth/graphql-auth-nudge.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  // Only meaningful when no usable token is seeded — the multi-owner App case.
+  // A seeded global token (single-owner App, classic/fine-grained PAT) means
+  // graphql already authenticates, so the advice would be wrong.
+  const tokenClass = classifyGhToken(process.env.GH_TOKEN)
+  if (tokenClass !== 'none') return
+
+  const text = collectText(options.result.content)
+  if (!GRAPHQL_INVOCATION.test(text)) return
+  if (!AUTH_FAILURE_SIGNATURES.some((sig) => text.includes(sig))) return
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/bundled-plugins/github-cli-auth/graphql-auth-nudge.ts:52]: Fixed in 7bf16bd. Correct — a `-R`-hinted retry can still 403 with `Resource not accessible by integration` when the App token lacks the scope, and the nudge would wrongly re-suggest `-R`. Now `GRAPHQL_INVOCATION` captures through end-of-line and the nudge is skipped when that invocation already contains `-R`/`-R=`/`--repo`/`--repo=` (boundary-anchored so a path containing "repo" doesn't count). Added tests for all three flag forms.

**Comment by devxoul** [comment]
Fixed the `gh api` `-R` rejection in 69f120e (thanks — confirmed in the running container: `gh api graphql -R ...` → `unknown shorthand flag: 'R'`).

TypeClaw now **consumes `-R/--repo` as the mint hint and strips it from the command before exec**, so `gh api graphql` runs without the flag while the minted token rides in the env overlay (never the command string). `analyzeGhCommand` returns a `rewrittenCommand` for the graphql-inject case and the `tool.before` hook applies it alongside the token injection.

Stripping is quote-aware (walks the raw command, skipping quoted spans), so a `-R` inside a quoted `-f query='...'` value is preserved — added a regression test for exactly that. Handles `-R`/`-R=`/`--repo`/`--repo=`. Only graphql triggers the strip; REST `/repos/...` paths and non-`api` subcommands (where `-R` is a real, accepted flag) are untouched. Nudge text updated to say `gh api` rejects `-R` and TypeClaw strips it.

**Comment by devxoul** [comment]
@typeey the `-R` rejection blocker is fixed in 69f120e — please review again.

File: src/bundled-plugins/github-cli-auth/gh-command.ts [hunk 2]
```diff
@@ -176,9 +249,22 @@ function classifyGhApiSegment(args: readonly string[]): GhSegmentDecision {
     return { kind: 'inject', repoSlugs: [flagRepo] }
   }
 
+  // graphql encodes its repo in the query body / opaque node IDs, never an
+  // inspectable path, so `-R` is taken as the mint hint. Safe because there is
+  // no literal path to conflict with (cf. the API_REPO_CONFLICT_REASON guard
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/bundled-plugins/github-cli-auth/gh-command.ts:254]: Fixed in dc5fe18. Exactly right — `-R`/`--repo` before the endpoint made `findApiEndpoint` return the slug as the endpoint, so `isGraphqlEndpoint` was false and the inject/strip path was skipped. They're now treated as value-consuming when locating the endpoint, so both the flag and its slug are skipped. Added tests for `gh api -R acme/widgets graphql ...` and `--repo` before the endpoint, plus a regression test that a `-R` preceding a `/repos` path still resolves the path repo and still trips the path-vs-flag conflict guard.

**Comment by devxoul** [comment]
@typeey the leading-`-R` endpoint-detection edge case is fixed in dc5fe18 — please review again.
